src/Redux/RocketsSlice/RocketSlice.js

```diff
@@ -0,0 +1,72 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+export const rocketItems = createAsyncThunk('rocket', async () => {
+  const response = await fetch(url);
+  const result = await response.json();
+
+  // Extract only the required fields from the JSON data
+  const rocketData = result.map((item) => ({
+    id: item.rocket_id,
+    name: item.rocket_name,
+    description: item.description,
+    image: item.flickr_images[0],
+  }));
+
+  return rocketData;
+});
+
+const initialState = {
+  RocketDatile: [],
+  loading: false,
+  error: null,
+  buttonText: 'reserve rocket',
+};
+
+const rocketSlice = createSlice({
+  name: 'rocket',
+  initialState,
+  reducers: {
+    reserveActive: (state, action) => {
+      const newRockets = state.RocketDatile.map((item) => {
+        if (item.name !== action.payload) return item;
+        return { ...item, reserved: true };
+      });
+
+      return {
+        ...state,
+        RocketDatile: newRockets,
+      };
+    },
+    cancelReserve: (state, action) => {
+      const newRockets = state.RocketDatile.map((item) => {
+        if (item.name !== action.payload) return item;
+        return { ...item, reserved: false };
+      });
+      return {
+        ...state,
+        RocketDatile: newRockets,
+      };
```

Comment:
- [ ]  It will be great to use the `ID` of the rocket instead of the name when you are canceling your reservation. This is because the` ID `doesn't change in the database but the name can be edited. If the name changes then the app might start to misbehave. Also, it is a major requirement for you to use the ID in this project ⬇️ 
![reservation by name](https://user-images.githubusercontent.com/32361963/233601715-7344c9a9-cea1-4945-a897-f0e52d83fc9c.png)